logic.js

```diff
@@ -36,429 +36,439 @@ http://ricostacruz.com/cheatsheets/umdjs.html
-      return (typeof rule === "string");
-    }
-    if(pattern === "array") {
-      // !logic test might be superfluous in JavaScript
-      return Array.isArray(rule) && ! jsonLogic.is_logic(rule);
-    }
 
-    if(jsonLogic.is_logic(pattern)) {
-      if(jsonLogic.is_logic(rule)) {
-        var pattern_op = jsonLogic.get_operator(pattern);
-        var rule_op = jsonLogic.get_operator(rule);
-
-        if(pattern_op === "@" || pattern_op === rule_op) {
-        // echo "\nOperators match, go deeper\n";
-          return jsonLogic.rule_like(
-            jsonLogic.get_values(rule, false),
-            jsonLogic.get_values(pattern, false)
-          );
+      if(jsonLogic.is_logic(pattern)) {
+        if(jsonLogic.is_logic(rule)) {
+          var pattern_op = jsonLogic.get_operator(pattern);
+          var rule_op = jsonLogic.get_operator(rule);
+
+          if(pattern_op === "@" || pattern_op === rule_op) {
+          // echo "\nOperators match, go deeper\n";
+            return jsonLogic.rule_like(
+              jsonLogic.get_values(rule, false),
+              jsonLogic.get_values(pattern, false)
+            );
+          }
         }
+        return false; // pattern is logic, rule isn't, can't be eq
       }
-      return false; // pattern is logic, rule isn't, can't be eq
-    }
 
-    if(Array.isArray(pattern)) {
-      if(Array.isArray(rule)) {
-        if(pattern.length !== rule.length) {
-          return false;
-        }
-        /*
-          Note, array order MATTERS, because we're using this array test logic to consider arguments, where order can matter. (e.g., + is commutative, but '-' or 'if' or 'var' are NOT)
-        */
-        for(var i = 0; i < pattern.length; i += 1) {
-          // If any fail, we fail
-          if( ! jsonLogic.rule_like(rule[i], pattern[i])) {
+      if(Array.isArray(pattern)) {
+        if(Array.isArray(rule)) {
+          if(pattern.length !== rule.length) {
             return false;
           }
+          /*
+            Note, array order MATTERS, because we're using this array test logic to consider arguments, where order can matter. (e.g., + is commutative, but '-' or 'if' or 'var' are NOT)
+          */
+          for(var i = 0; i < pattern.length; i += 1) {
+            // If any fail, we fail
+            if( ! jsonLogic.rule_like(rule[i], pattern[i])) {
+              return false;
+            }
+          }
+          return true; // If they *all* passed, we pass
+        }else{
+          return false; // Pattern is array, rule isn't
         }
-        return true; // If they *all* passed, we pass
-      }else{
-        return false; // Pattern is array, rule isn't
       }
-    }
 
-    // Not logic, not array, not a === match for rule.
-    return false;
-  };
+      // Not logic, not array, not a === match for rule.
+      return false;
+    };
+
+    return jsonLogic;
+  }
+
+
+  var retval = new JSONLogic();
+
+  retval.JSONLogic = JSONLogic;
+
+  return retval;
```

Comment:
Lines 468 - 472 are the only real "change" to this file.  They simply:

- Create an instance of JSONLogic.  (ie: by invoking the factory function)
- Inject the factory function into this instance.  (So that developers can create separate isolated instances of JSONLogic)